### ISSUE TITLE
prop-types false positive in block arrow function

### ISSUE THREAD

**Issue by billyjanitsch** [yannickcr/eslint-plugin-react#487]
``` js
import {PropTypes} from 'react'

export const MyComponentA = props =>
  <div onClick={() => {props.test()}} />

MyComponentA.propTypes = {
  test: PropTypes.func,
}

export const MyComponentB = props =>
  <div onClick={() => {props.otherTest()}} /> // 'otherTest' is missing in props validation

MyComponentB.propTypes = {
  otherTest: PropTypes.func,
}
```

Expected: `props.otherTest` should not throw because it is correctly validated.

eslint: 2.2.0
eslint-plugin-react: 4.1.0

**Comment by yannickcr** [comment]
Seems to be the same problem as #485

**Comment by billyjanitsch** [comment]
Thanks for all of the quick fixes @yannickcr!

**Comment by yannickcr** [comment]
You're welcome :) Don't hesitate to reopen the issue if it's not fixed in next release.
